core/runtime/src/main/java/io/quarkus/runtime/configuration/HyphenateEnumConverter.java

```diff
@@ -13,6 +13,7 @@
 /**
  * A converter for hyphenated enums.
  */
+@Deprecated
```

Comment:
Perhaps indicate the recommended way to use it? Also `forRemoval=true` may be interesting here